src/main/java/org/assertj/core/presentation/CompositeRepresentation.java

```diff
@@ -0,0 +1,76 @@
+package org.assertj.core.presentation;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.presentation.StandardRepresentation.STANDARD_REPRESENTATION;
+
+public class CompositeRepresentation implements Representation {
+  protected Map<String, Representation> representations;
+
+  /*Initialize the CompositeRepresentation according to a certain correspondence of the "name of representation" and "representation" itself.*/
+  public CompositeRepresentation(Map<String, Representation> representations) {
+    this.representations = representations;
+  }
+
+  /*Initialize the CompositeRepresentation when the user does not have the correspondence of the "name of representation" and "representation" itself yet.*/
+  public CompositeRepresentation() {
+    representations = new HashMap<>();
+  }
+
+  /*Implementation of the `toString` method.*/
+  @Override
+  public String toStringOf(Object object) {
+    for (Map.Entry<String, Representation> entry : representations.entrySet()) {
+      String value = entry.getValue().toStringOf(object);
+      if (value != null)
+        return value;
+    }
+    return STANDARD_REPRESENTATION.toStringOf(object);
+  }
+
+  /*Implementation of the `toString` method when a specific representation is desired.*/
+  public String toStringOf(Object object, String representationName) {
```

Comment:
I'm not sure we have a use case for this one, can you show an example that would justify adding it?

Reply:
Thank you very much for your suggestions! As for this function, if there are many representations for one object, for example, we want to represent a number in binary or hexadecimal way in different situations, we can use `key` of the `representationsByName` to indicate which representation we want to have. So this function was added.

Reply:
If the example I imagine does not exist, I suppose `List` is better than `Map` in this situation, and `toStringOf(Object object, String representationName)` can be removed.